### PR TITLE
core/device: fix devlink handling 

### PR DESCRIPTION
**Pull request by yuwata**
~~After #23507.~~

### CODE REVIEW

File: src/core/device.c [hunk 1]
```diff
@@ -1001,27 +1102,49 @@ static int device_dispatch_io(sd_device_monitor *monitor, sd_device *dev, void *
...
+                }
 
+                /* Add additional units for all symlinks */
+                (void) device_setup_devlink_units(m, dev, &ready_devlinks);
+        }
+
+        if (!IN_SET(action, SD_DEVICE_ADD, SD_DEVICE_REMOVE, SD_DEVICE_MOVE))
+                device_propagate_reload_by_sysfs(m, sysfs);
+
+        STRV_FOREACH(devlink, ready_devlinks)
```
Comment (msekletar) [inline src/core/device.c]: This block is run for all event types. Is that what we want? It seems to me we don't want to propagate reloads on REMOVE events, right?

Comment (yuwata) [reply src/core/device.c]: To be frank, I have no idea about that. What does `manager_propagate_reload()` do?? It is difficult to understand transaction.c and job.c...

File: src/core/manager.h [hunk 2]
```diff
@@ -406,6 +406,8 @@ struct Manager {
         char *switch_root;
         char *switch_root_init;
 
+        bool switching_root;
```
Comment (poettering) [inline src/core/manager.h:410]: why have that as a separate boolean? Why isn't `m->objective == MANAGER_SWITCH_ROOT` enough?

Comment (yuwata) [reply src/core/manager.h:410]: `MANAGER_SWITCH_ROOT` is set before the system is going to switch root. The `switching_root` flag is set the PID1 started after switching root. Maybe, the name of the boolean flag should be changed. Do you have any better idea?

Comment (poettering) [reply src/core/manager.h:410]: might suffice to add a big comment here, explaining that this is set both before *and* after the switch root, in the old and the new PID 1 process.

File: src/core/device.c [hunk 3]
```diff
@@ -719,9 +712,111 @@ static bool device_is_ready(sd_device *dev) {
...
+        assert(devlink);
+
+        if (sd_device_new_from_devname(&dev, devlink) < 0 || !device_is_ready(dev)) {
+                /* the devlink is already removed or not ready */
+                device_update_found_by_name(m, devlink, 0, DEVICE_FOUND_UDEV);
+                *ret = NULL;
+                return 0; /* not ready */
+        }
+
+        r = device_setup_unit(m, dev, devlink, false);
```
Comment (poettering) [inline src/core/device.c]: maybe do the `/* param_name= */` dance on the fourth agument

**Comment by yuwata** [comment]
@poettering Thank you for the review. All comments are addressed. Upgrading the green label.

**Pull request merged**